gitCore/jGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java

```diff
@@ -145,19 +146,24 @@ private GitCoreCommit convertObjectIdToGitCoreCommit(ObjectId objectId) throws G
     return withRevWalk(walk -> new GitCoreCommit(walk.parseCommit(objectId)));
   }
 
-  private Option<GitCoreCommit> convertRevisionToGitCoreCommit(String revision) throws GitCoreException {
-    return convertRevisionToObjectId(revision).flatMap(
-        objectId -> Try.of(() -> withRevWalkUnchecked(walk -> new GitCoreCommit(walk.parseCommit(objectId)))).toOption());
+  private @Nullable GitCoreCommit convertRevisionToGitCoreCommit(String revision) throws GitCoreException {
+    val objectId = convertRevisionToObjectId(revision);
+    return objectId != null
+        ? Try.of(() -> withRevWalkUnchecked(walk -> new GitCoreCommit(walk.parseCommit(objectId)))).getOrNull()
+        : null;
   }
 
   private ObjectId convertExistingRevisionToObjectId(String revision) throws GitCoreException {
-    return convertRevisionToObjectId(revision)
-        .getOrElseThrow(() -> new GitCoreNoSuchRevisionException("Commit '${revision}' does not exist in this repository"));
+    val objectId = convertRevisionToObjectId(revision);
+    if (objectId == null) {
+      new GitCoreNoSuchRevisionException("Commit '${revision}' does not exist in this repository");
```

Comment:
Whoops exception **not** thrown, pls check for other such places

Reply:
Pls check if this fixes the problem with Lombok somehow 🤔

Reply:
it didn't 😔

Reply:
Okay pls investigate yourself for a moment (try to cut out a minimal repro), I can help later in the day/tmrw